From 0c1272cc7c72dfe0ef66be8f283cf67c74b58586 Mon Sep 17 00:00:00 2001
From: Nicholas Piggin <npiggin@gmail.com>
Date: Sun, 6 May 2018 17:29:49 +1000
Subject: [PATCH] osdep: powerpc64 align memory to allow 2MB radix THP page
 tables

This allows KVM with the Book3S radix MMU mode to take advantage of
THP and install larger pages in the partition scope page tables (the
host translation).

Signed-off-by: Nicholas Piggin <npiggin@gmail.com>
Signed-off-by: David Gibson <david@gibson.dropbear.id.au>

Author: Christian Ehrhardt <christian.ehrhardt@canonical.com>
Original-Author: Nicholas Piggin <npiggin@gmail.com>
Origin: backport, https://git.qemu.org/?p=qemu.git;a=commit;h=0c1272cc7c72dfe0ef66be8f283cf67c74b58586
Bug-Ubuntu: https://bugs.launchpad.net/bugs/1781526
Last-Update: 2018-06-16

---
 include/qemu/osdep.h | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/include/qemu/osdep.h b/include/qemu/osdep.h
index afc28e5903..9ed62423c0 100644
--- a/include/qemu/osdep.h
+++ b/include/qemu/osdep.h
@@ -367,7 +367,8 @@ void qemu_anon_ram_free(void *ptr, size_t size);
 #endif
 
 #if defined(__linux__) && \
-    (defined(__x86_64__) || defined(__arm__) || defined(__aarch64__))
+    (defined(__x86_64__) || defined(__arm__) || defined(__aarch64__) \
+     || defined(__powerpc64__))
    /* Use 2 MiB alignment so transparent hugepages can be used by KVM.
       Valgrind does not support alignments larger than 1 MiB,
       therefore we need special code which handles running on Valgrind. */
-- 
2.17.1

